### PR TITLE
Update devel doc WRT Ignition remediation

### PR DESCRIPTION
**Pull request by ggbecker**
#### Description:

- Add developer documentation with regards to Ignition remediation.
- Enable template for Ignition remediation.

- Fix small leftovers of legacy rule id's format.

### CODE REVIEW

File: docs/manual/developer_guide.adoc [hunk 1]
```diff
@@ -815,6 +815,7 @@ The <tt>/var/tmp</tt> directory is a world-writable directory used for temporary
 +
 When deciding on severity levels, it is best to follow the following guidelines:
 .Table Vulnerability Severity Category Code Definitions
++
```
Comment (ggbecker) [inline docs/manual/developer_guide.adoc]: Display rich format option from Github shows a lot of changes, but this `+` was missing and now https://github.com/ComplianceAsCode/content/pull/5144/files#diff-bcd4620adacf98f7425fd2f913933222R834 shows correctly under the table.

File: docs/manual/developer_guide.adoc [hunk 2]
```diff
@@ -833,7 +834,7 @@ When deciding on severity levels, it is best to follow the following guidelines:
...
 * `ocil`: Defines asserting statements to check whether or not the rule is valid.
-* `ocil_clause`: This attribute contains the statement which describes how to determine whether the statement is true or false. Check out `encrypt_partitions.rule` in `linux_os/guide/system/software/disk_partitioning/`: this contains a `partitions do not have a type of crypto_LUKS` value for `ocil_clause`. This clause is prefixed with the phrase "It is the case that".
+* `ocil_clause`: This attribute contains the statement which describes how to determine whether the statement is true or false. Check out `rule.yml` in `linux_os/guide/system/software/disk_partitioning/encryp_partitions/`: this contains a `partitions do not have a type of crypto_LUKS` value for `ocil_clause`. This clause is prefixed with the phrase "It is the case that".
```
Comment (jan-cerny) [inline docs/manual/developer_guide.adoc]: encrypt

Comment (ggbecker) [reply docs/manual/developer_guide.adoc]: fixed

File: ssg/templates.py [hunk 3]
```diff
@@ -6,14 +6,15 @@
...
-languages = ["anaconda", "ansible", "bash", "oval", "puppet"]
+languages = ["anaconda", "ansible", "bash", "oval", "puppet", "ignition"]
 
 lang_to_ext_map = {
     "anaconda": ".anaconda",
     "ansible": ".yml",
     "bash": ".sh",
     "oval": ".xml",
     "puppet": ".pp"
+    "ignition": ".yml"
```
Comment (jan-cerny) [inline ssg/templates.py]: missing comma after `.pp`.

Comment (ggbecker) [reply ssg/templates.py]: fixed. Thanks for noticing it

**Pull request merged**